Fixes empty response issue #1

matus-usiak opened the pull request:
### Motivation

Fixes #1 

### Checklist

- [✓] Code follows coding conventions held in this repo
- [ ] Automated tests have been added
- [ ] Tests are passing
- [✓] Docs have been updated (if applicable)
- [✓] Temporary settings (e.g. variables used during development and testing) have been reverted to defaults

### How to test

- try accessing an unpublished item
- access for example ~/Denver/Speakers/cj005

matus-usiak committed:
Commit to address the empty response issue

matus-usiak committed:
the response doesn't return just the first item

matus666 commented:
weeeell..returning all items in the response (baseController line 30) has not gone too well ;/ 
it works fine for landing page (http://localhost:57555) but not for home (http://localhost:57555/Denver) then again it works for speakers (http://localhost:57555/Denver/Speakers) but not for the individual speaker (http://localhost:57555/Denver/Speakers/bo001)

so I'll try to address that in a new pull request

kentico-michaelb reviewed (changes requested):
I think a better approach would be to leverage .Net Core's _UseExceptionHandler_ mentioned in the "The UseExceptionHandler Middleware" section here: https://www.red-gate.com/simple-talk/dotnet/net-development/asp-net-core-3-0-exception-handling/  .  This will allow us to have "friendly" custom error pages as well.

I think this is the approach used by DevRel in the .Net Core Boilerplate:
**Startup.cs:**
https://github.com/Kentico/kontent-boilerplate-net/blob/master/src/content/Kentico.Kontent.Boilerplate/Startup.cs#L68
**Error Controller**
https://github.com/Kentico/kontent-boilerplate-net/blob/master/src/content/Kentico.Kontent.Boilerplate/Controllers/ErrorController.cs
**Views**
https://github.com/Kentico/kontent-boilerplate-net/tree/master/src/content/Kentico.Kontent.Boilerplate/Views/Error

Lets give this a shot in this same branch/pull. This approach may also resolve the other issues you discovered in your testing as well.

matus-usiak committed:
this seems like it's doing the job

kentico-michaelb reviewed (changes requested):
The high-level behavior seems to be correct now (registering the handlers in Startup and creating the Error controller + views), but attempting to use the navigation from the error pages gets the user stuck in an error loop because a location does not exist in the error URL.

Example:
1. Enter an incorrect URL to land on the 404 page
2. Click a navigation link form the top menu

Result: a URL like "http://localhost:57555/Venue" results in a 500 error since the location parameter is not being set, and the Venue Controller's call to the API fails.

The _easiest_ solution would be to set the error views Layout = null and provide a JavaScript "back" link to return them to the previous page they were browsing so users don't have the option to use the broken navigation links.  

The _better_ solution could be to pass the current location to the error controller's viewbag (similar to the other controllers) if it exists that way the navigation menu has the correct URL.  If the location isn't present, it means the user has not selected a conference and should only be able to return to the landing page or back to the previous page they were on instead of seeing a full navigation.

Further changes:
1. Remove the old behavior that is currently commented out. 
2. Correct the typo in the GeneralError.cshtml file ("whops" = "whoops")

matus-usiak committed:
1. removed commented out behavior
2. corrected type
3. added the 'back' button to the errror pages - 404 and whoops

matus-usiak committed:
just removed some more unused code
